proof/refine/AARCH64/Invariants_H.thy

```diff
@@ -838,10 +863,15 @@ where
 | "runnable' (Structures_H.BlockedOnSend a b c d e) = False"
 | "runnable' (Structures_H.BlockedOnNotification x) = False"
 
-definition
-  inQ :: "domain \<Rightarrow> priority \<Rightarrow> tcb \<Rightarrow> bool"
-where
- "inQ d p tcb \<equiv> tcbQueued tcb \<and> tcbPriority tcb = p \<and> tcbDomain tcb = d"
+definition inQ :: "domain \<Rightarrow> priority \<Rightarrow> tcb \<Rightarrow> bool" where
+  "inQ d p tcb \<equiv> tcbQueued tcb \<and> tcbPriority tcb = p \<and> tcbDomain tcb = d"
+
+lemma inQ_implies_tcbQueueds_of:
+  "(inQ domain priority |< tcbs_of' s') tcbPtr \<Longrightarrow> (tcbQueued |< tcbs_of' s') tcbPtr"
+  by (clarsimp simp: opt_map_def opt_pred_def inQ_def split: option.splits)
+
+defs ready_qs_runnable_def:
+  "ready_qs_runnable s \<equiv> \<forall>t. obj_at' tcbQueued t s \<longrightarrow> st_tcb_at' runnable' t s"
```

Comment:
I'm a bit confused, why does this have a separate `_def` here? does it clobber the existing one?

Reply:
Oh it's not a style/naming issue, I'm confused as to what it's doing. It looks like it's clobbering the existing definition somehow, or maybe there was only a `consts'` and this provides the definition?

Reply:
Yes it was just a `consts'` and this provides the definition